storm/src/main/storm/mesos/schedulers/OfferResources.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package storm.mesos.schedulers;
+
+import com.google.common.base.Joiner;
+import org.apache.log4j.Logger;
+import org.apache.mesos.Protos;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OfferResources {
+  private final Logger log = Logger.getLogger(OfferResources.class);
+
+  public enum ResourceType {
+    CPU,
+    MEM,
+    PORT
+  }
+
+  private class PortRange {
+    public long begin;
+    public long end;
+
+    public PortRange(long begin, long end) {
+      this.begin = begin;
+      this.end = end;
+    }
+  }
+
+  Protos.Offer offer;
+  Protos.OfferID offerId;
+  String hostName;
+  double mem;
+  double cpu;
+
+  List<PortRange> portRanges = new ArrayList<>();
+
+  public void addPortRanges(Protos.Value.Ranges ranges) {
+    for (Protos.Value.Range r : ranges.getRangeList()) {
+      this.portRanges.add(new PortRange(r.getBegin(), r.getEnd()));
+    }
+  }
+
+  public OfferResources(Protos.Offer offer) {
+    this.offer = offer;
+    this.offerId = offer.getId();
+    double offerMem = 0;
+    double offerCpu = 0;
+    Protos.Value.Ranges portRanges = null;
+
+    String hostName = offer.getHostname();
+    for (Protos.Resource r : offer.getResourcesList()) {
+      if (r.getName().equals("cpus")) {
+        offerCpu = r.getScalar().getValue();
+      } else if (r.getName().equals("mem")) {
+        offerMem = r.getScalar().getValue();
+      } else if (r.getName().equals("ports")) {
+        portRanges = r.getRanges();
+      }
+    }
+
+    this.hostName = hostName;
+    this.mem = offerMem;
+    this.cpu = offerCpu;
+    if ((portRanges != null) && (!portRanges.getRangeList().isEmpty())) {
+      this.addPortRanges(portRanges);
+    }
+  }
+
```

Comment:
@dsKarthick I looked around a bit, it's sort of implicit the first relevant sections is `getResources` in [MesosNimbus.java 300-318](https://github.com/mesos/storm/blob/master/storm/src/main/storm/mesos/MesosNimbus.java#L300-L318). You replaced that method with the constructor above but note the deferences of `+=` and `=`.  By adding the values you get the full value of all Scalars with name `cpus` in the offer where one part is the will have a role the other part won't.  
The second place you see this is in `getResourceScalar` [MesosNimbus.java#L391-423](https://github.com/mesos/storm/blob/master/storm/src/main/storm/mesos/MesosNimbus.java#L391-L423), here we loop through the resources pulling out the amount needed as we find resources of the appropriate name and return a list of Resources of Type `SCALAR` and Name `cpus`.  This is important because if you have one resource with 4 cpus and another with 2, and you tried to create a task info with one Resource of name `cpus` with value 6, you'd get a task failure.  This may be worth commenting.

Anyway, I think since you're still using `getScalarResource` to create the resources, you may be OK just updating the constructor.  I'll try to review the relevant pieces of your code tomorrow to verify.